mlflow/tensorflow/__init__.py

```diff
@@ -117,10 +110,13 @@ def get_default_conda_env():
 @keyword_only
 @format_docstring(LOG_MODEL_PARAM_DOCS.format(package_name=FLAVOR_NAME))
 def log_model(
-    tf_saved_model_dir,
-    tf_meta_graph_tags,
-    tf_signature_def_key,
+    *,
     artifact_path,
+    tf_saved_model_dir=None,
```

Comment:
Is this still the easiest / recommended way to save a tensor flow model, or could we improve the API here too? Ie is it possible to accept an estimator object or something similar instead?

Reply:
> is it possible to accept an estimator object or something similar instead

Sounds a good idea. what about do it in follow-up task ?

Reply:
I think we should investigate whether we could improve the API first. We might not need the `keras_model` argument.

Reply:
I think the reason current `log_model` API providing `tf_saved_model_dir` for tensorflow estimator is:
current TF autologging only log model when user code calls `tf_estimator.export_saved_model`, and in autologging patching code, it does not call `export_saved_model`, but instead, hook the method and when user code calls it, capture the output result (i.e. the `tf_saved_model_dir` ) and log the model.
If we change log_model to accept `estimator` object as argument, we need to change above desgin.

Reply:
We can change `log_model` / `save_model` to accept TF estimator as argument to replace `tf_saved_model_dir`, but we need to add a new argument `serving_input_receiver_fn` (required by `estimator.export_saved_model`)